code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -31,85 +68,207 @@
+	var/spawn_delay = 300
+	var/activated = 0
+	var/snoozing = 0 //If set to 1, it will be prevented from spawning bots, unless it spots an enemy.
+	attack_emote = "focuses on"
+
+/mob/living/simple_animal/hostile/hivebotbeacon/toxic
+	projectiletype = /obj/item/projectile/beam/hivebotbeam
+	projectilesound = 'sound/weapons/laser3.ogg'
+	rapid = 0
 
-/mob/living/simple_animal/hostile/hivebot/tele/New()
+/mob/living/simple_animal/hostile/hivebotbeacon/incendiary
+	projectiletype = /obj/item/projectile/beam/hivebotincendiary
+	projectilesound = 'sound/weapons/plasma_cutter.ogg'
+	rapid = 0
+
+/mob/living/simple_animal/hostile/hivebotbeacon/New()
 	..()
-	var/datum/effect/effect/system/smoke_spread/smoke = new /datum/effect/effect/system/smoke_spread()
-	smoke.set_up(5, 0, src.loc)
-	smoke.start()
-	visible_message("<span class='danger'>\The [src] warps in!</span>")
+	var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread()
+	S.set_up(5, 0, src.loc)
+	S.start()
+	icon_state = "hivebotbeacon_off"
+	visible_message("<span class='danger'>\ [src] warps in!</span>")
 	playsound(src.loc, 'sound/effects/EMPulse.ogg', 25, 1)
 
-/mob/living/simple_animal/hostile/hivebot/tele/proc/warpbots()
+/mob/living/simple_animal/hostile/hivebotbeacon/death()
+	..(null,"blows apart and erupts in a cloud of noxious smoke!")
+	var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread
+	S.set_up(5, 0, src.loc)
+	S.start()
+	new /obj/effect/decal/cleanable/greenglow(src.loc)
+	var/T = get_turf(src)
+	new /obj/effect/gibspawner/robot(T)
+	spark(T, 3, alldirs)
+	for (var/mob/living/simple_animal/hostile/hivebot/latest_child in linked_bots)
+		latest_child.linked_parent = null
+	linked_bots.Cut()
+	qdel(src)
+	return
+
+/mob/living/simple_animal/hostile/hivebotbeacon/think()
+	. =..()
+	if(stance != HOSTILE_STANCE_IDLE && activated == 0)
+		visible_message("<span class='warning'>\ [src] suddenly activates!</span>")
+		icon_state = "hivebotbeacon_raising"
+		spawn(16)
+		icon_state = "hivebotbeacon_active"
+		activated = 1
+		snoozing = 0
+	else
+		if(activated == 1)
+			icon_state = "hivebotbeacon_active"
+
+/mob/living/simple_animal/hostile/hivebotbeacon/emp_act(severity)
+	LoseTarget()
+	stance = HOSTILE_STANCE_TIRED
+	addtimer(CALLBACK(src, .proc/wakeup), 200)
+	if(severity == 1.0)
+		apply_damage(80)
+	icon_state = "hivebotbeacon_off"
+	activated = -1
+
+/mob/living/simple_animal/hostile/hivebotbeacon/proc/wakeup()
+	stance = HOSTILE_STANCE_IDLE
+	activated = 0
 
+/mob/living/simple_animal/hostile/hivebotbeacon/proc/warpbots()
 	if(!bot_amt)
+		visible_message("<b>The Beacon</b> disappears in a cloud of smoke!")
+		playsound(src.loc, 'sound/effects/teleport.ogg', 25, 1)
+		var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread()
+		S.set_up(5, 0, src.loc)
+		S.start()
+		for (var/mob/living/simple_animal/hostile/hivebot/latest_child in linked_bots)

```

Comment:
See above note. IF this code (lines 230 - 232) goes into `Destroy()`, then you don't have to duplicate it here.